hoomd/mpcd/ParticleData.cc

```diff
@@ -416,6 +457,128 @@ void mpcd::ParticleData::initializeRandom(unsigned int N, const BoxDim& local_bo
         }
     }
 
+/*!
+ * \param density Density of particles inside sphere
+ * \param R Radius of sphere 
+ * \param local_box Local simulation box
+ * \param kT Temperature (in energy units)
+ * \param seed Random seed
+ * \param ndimensions Dimensionality
+ */
+void mpcd::ParticleData::initializeRandomSphere(Scalar density, Scalar R, const BoxDim& local_box, Scalar kT, unsigned int seed, unsigned int ndimensions)
+    {
+    // only one particle type is supported for this construction method
+    m_type_mapping.clear();
+    m_type_mapping.push_back("A");
+    // default particle mass is 1.0
+    m_mass = Scalar(1.0);
+
+    // throw runtime error if radius is negative or zero
+    if (R <= Scalar(0.0))
+        {
+        throw std::runtime_error("Radius of sphere can't be negative or zero");
+        }
+
+    // Calculating Ndraw - number of particles that should be inside the local box according to density
+    unsigned int Ndraw = round(density*local_box.getVolume());
+
+    // Allocating an Array for storing the random drawn positions
+    GPUArray<Scalar4> pos_draw(Ndraw, m_exec_conf);
+    m_pos_draw.swap(pos_draw);
+    ArrayHandle<Scalar4> h_pos_draw(m_pos_draw, access_location::host, access_mode::overwrite);
+
```

Comment:
We can use a `std::vector` here since this is CPU code, and reserve space to push back into to know how many we've drawn:
```suggestion
    std::vector<Scalar4> pos_draw;
    pos_draw.reserve(Ndraw);
```